cmd/demo/demo.go

```diff
@@ -0,0 +1,340 @@
+        {{.}}: 
+          secret-1: 0
+          secret-2: 1
+      kubernetes:
+        ns-a:
+          secret-a:
+            key-a: -1
+        ns-b:
+          secret-b:
+            key-b: -1
+    `
+	// substitute fixture project to o.gsmProject
+	temp := template.Must(template.New("config").Parse(fixtureConfig))
+	fixtureBuffer := new(bytes.Buffer)
+	temp.Execute(fixtureBuffer, o.gsmProject)
+
+	fixture, err := tests.NewFixture(fixtureBuffer.Bytes())
+	if err != nil {
+		klog.Fatalf("Fail to parse fixture: %s", err)
+	}
+
+	err = fixture.Setup(testClient)
+	if err != nil {
+		klog.Fatalf("Fail to setup fixture: %s", err)
+	}
+
+	defer fixture.Teardown(testClient)
+
+	// start controller and logger
+	stopLogger := make(chan struct{})
+	stopController := make(chan struct{})
+	go logger.Start(stopLogger)
+	go controller.Start(stopController)
+
+	time.Sleep(time.Duration(o.duration) * time.Millisecond)
+	stopLogger <- struct{}{}
+	stopController <- struct{}{}
+
+	logger.ShowResults()
+}
+
+type SecretSyncLogger struct {
+	Client       client.Interface
+	Agent        *config.Agent
+	PollPeriod   time.Duration
+	SwitchPeriod time.Duration
+	LogData      map[string]*logData
+}
+
+type logData struct {
+	GSMSecretLog []string
+	K8sSecretLog []string
+	Time         []float64
+}
+
+func (l *SecretSyncLogger) Start(stopChan <-chan struct{}) error {
+	logChan := make(chan struct{})
+	switchChan := make(chan struct{})
+
+	startTime := time.Now()
+
+	// controller period for logging
+	go func() {
+		for {
+			logChan <- struct{}{}
+			time.Sleep(l.PollPeriod)
+		}
+	}()
+
+	// controller period for source secret value switch
+	go func() {
+		for {
+			switchChan <- struct{}{}
+			time.Sleep(l.SwitchPeriod)
+		}
+	}()
+
+	for {
+		select {
+		case <-stopChan:
+			klog.V(2).Info("Stop signal received. Quitting logger...")
+			return nil
+		case <-logChan:
+			for _, spec := range l.Agent.Config().Specs {
+
+				d, ok := l.LogData[spec.String()]
+				if !ok {
+					d = &logData{}
+					l.LogData[spec.String()] = d
+				}
+
+				// get secret values
+				srcData, err := l.Client.GetSecretManagerSecretValue(spec.Source.Project, spec.Source.Secret)
+				if err != nil {
+					klog.Errorf("Secret log failed for %s: %s", spec, err)
+				}
+
+				destData, err := l.Client.GetKubernetesSecretValue(spec.Destination.Namespace, spec.Destination.Secret, spec.Destination.Key)
+				if err != nil {
+					klog.Errorf("Secret log failed for %s: %s", spec, err)
+				}
+
+				// append to log
+				d.Append(time.Now().Sub(startTime), srcData, destData)
+			}
+		case <-switchChan:
+			for _, spec := range l.Agent.Config().Specs {
+				prev, _ := l.Client.GetSecretManagerSecretValue(spec.Source.Project, spec.Source.Secret)
+				next, _ := strconv.Atoi(string(prev))
+
+				// secret value swaps between 0 and 1 every SwitchPeriod
+				next = 1 - next
+				l.Client.UpsertSecretManagerSecret(spec.Source.Project, spec.Source.Secret, []byte(strconv.Itoa(next)))
```

Comment:
This seems brittle.  The comment only holds true given certain initial conditions.  Consider being more explicit?

```golang
next := "0"
if prev == next {
  next = "1"
}
```

Reply:
Yeah this explicit form is better. Modified.